examples/js/exporters/USDZExporter.js

```diff
@@ -13,23 +17,37 @@
 
 					const geometry = object.geometry;
 					const material = object.material;
-					materials[ material.uuid ] = material;
-					if ( material.map !== null ) textures[ material.map.uuid ] = material.map;
-					if ( material.normalMap !== null ) textures[ material.normalMap.uuid ] = material.normalMap;
-					if ( material.aoMap !== null ) textures[ material.aoMap.uuid ] = material.aoMap;
-					if ( material.roughnessMap !== null ) textures[ material.roughnessMap.uuid ] = material.roughnessMap;
-					if ( material.metalnessMap !== null ) textures[ material.metalnessMap.uuid ] = material.metalnessMap;
-					if ( material.emissiveMap !== null ) textures[ material.emissiveMap.uuid ] = material.emissiveMap;
-					output += buildXform( object, buildMesh( geometry, material ) );
+
+					if ( ! ( geometry.uuid in geometries ) ) {
+
+						geometries[ geometry.uuid ] = geometry;
+
+					}
+
+					if ( ! ( material.uuid in materials ) ) {
+
+						materials[ material.uuid ] = material;
+						if ( material.map !== null ) textures[ material.map.uuid ] = material.map;
+						if ( material.normalMap !== null ) textures[ material.normalMap.uuid ] = material.normalMap;
+						if ( material.aoMap !== null ) textures[ material.aoMap.uuid ] = material.aoMap;
+						if ( material.roughnessMap !== null ) textures[ material.roughnessMap.uuid ] = material.roughnessMap;
+						if ( material.metalnessMap !== null ) textures[ material.metalnessMap.uuid ] = material.metalnessMap;
+						if ( material.emissiveMap !== null ) textures[ material.emissiveMap.uuid ] = material.emissiveMap;
+
+					}
+
+					const referencedMesh = `prepend references = @./${geometryFileName}@</Geometry_${geometry.id}>`;
+					const referencedMaterial = `rel material:binding = </Materials/Material_${material.id}>`;
+					output += buildXform( object, referencedMesh, referencedMaterial );
 
 				}
 
 			} );
 			output += buildMaterials( materials );
 			output += buildTextures( textures );
-			const files = {
-				'model.usda': fflate.strToU8( output )
-			};
+			files[ modelFileName ] = fflate.strToU8( output );
+			output = null;
```

Comment:
This is to make the browser to deallocate memory, right?

Reply:
Yes. Most of the memory will be allocated during geometry file creation. So we are clearing "output" string right before it.